x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java

```diff
@@ -0,0 +1,360 @@
+            result.append(exp);
+        }
+
+        return result.toString();
+    }
+
+    private static String booleanExpression(List<Column> previousOutput) {
+        // TODO LIKE, RLIKE, functions etc.
+        return switch (randomIntBetween(0, 3)) {
+            case 0 -> {
+                String field = randomNumericField(previousOutput);
+                if (field == null) {
+                    yield null;
+                }
+                yield field + mathCompareOperator() + " 50";
+            }
+            case 1 -> "true";
+            default -> "false";
+        };
+    }
+
+    private static String mathCompareOperator() {
+        return switch (randomIntBetween(0, 5)) {
+            case 0 -> "==";
+            case 1 -> ">";
+            case 2 -> ">=";
+            case 3 -> "<";
+            case 4 -> "<=";
+            default -> "!=";
+        };
+    }
+
+    private static String enrich(List<Column> previousOutput, List<CsvTestsDataLoader.EnrichConfig> policies) {
+        String field = randomStringField(previousOutput);
+        if (field == null || policies.isEmpty()) {
+            return "";
+        }
+        // TODO add WITH
+        return " | enrich " + randomFrom(policies).policyName() + " on " + field;
+    }
+
+    private static String grok(List<Column> previousOutput) {
+        String field = randomStringField(previousOutput);
+        if (field == null) {
+            return "";// no strings to grok, just skip
+        }
+        StringBuilder result = new StringBuilder(" | grok ");
+        result.append(field);
+        result.append(" \"");
+        for (int i = 0; i < randomIntBetween(1, 3); i++) {
+            if (i > 0) {
+                result.append(" ");
+            }
+            result.append("%{WORD:");
+            if (randomBoolean()) {
+                result.append(randomAlphaOfLength(5));
+            } else {
+                result.append(randomName(previousOutput));
+            }
+            result.append("}");
+        }
+        result.append("\"");
+        return result.toString();
+    }
+
+    private static String dissect(List<Column> previousOutput) {
+        String field = randomStringField(previousOutput);
+        if (field == null) {
+            return "";// no strings to dissect, just skip
+        }
+        StringBuilder result = new StringBuilder(" | dissect ");
+        result.append(field);
+        result.append(" \"");
+        for (int i = 0; i < randomIntBetween(1, 3); i++) {
+            if (i > 0) {
+                result.append(" ");
+            }
+            result.append("%{");
+            if (randomBoolean()) {
+                result.append(randomAlphaOfLength(5));
+            } else {
+                result.append(randomName(previousOutput));
+            }
+            result.append("}");
+        }
+        result.append("\"");
+        return result.toString();
+    }
+
+    private static String keep(List<Column> previousOutput) {
```

Comment:
Here, you could easily (I hope) improve it by adding `*` or `fieldname*` or `*fieldname`.